src/commands/importFunctionApp/importFunctionApp.ts

```diff
@@ -0,0 +1,319 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ApiContract, BackendContract, BackendCredentialsContract, OperationCollection, OperationContract, PropertyContract } from "azure-arm-apimanagement/lib/models";
+import { WebSiteManagementClient } from "azure-arm-website";
+import { Site, WebAppCollection } from "azure-arm-website/lib/models";
+import { ProgressLocation, window } from "vscode";
+import xml = require("xml");
+import { FunctionAppService } from "../../Azure/WebApp/FunctionAppService";
+import { IFunctionContract } from "../../Azure/WebApp/IFunctionContract";
+import * as Constants from "../../constants";
+import { ApisTreeItem } from "../../explorer/ApisTreeItem";
+import { ApiTreeItem } from "../../explorer/ApiTreeItem";
+import { ServiceTreeItem } from "../../explorer/ServiceTreeItem";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+import { apiUtil } from "../../utils/apiUtil";
+import { nonNullOrEmptyValue } from "../../utils/nonNull";
+import { parseUrlTemplate } from "./parseUrlTemplate";
+
+export async function importFunctionAppToApi(node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue);
+    }
+
+    ext.outputChannel.show();
+    ext.outputChannel.appendLine(localize("importFunctionApp", `Import function app to api ${node.root.apiName} started...`));
+
+    const funcAppService = new FunctionAppService();
```

Comment:
line 31 to 38 are common for both functions importFunctionAppToApi importFunctionApp please refactor

Reply:
Try keep the code DRY ( DO NOT REPEAT YOURSELF) 

Reply:
The reason I put these common lines here is that if I put them in the same function, I have to return 4 values at which is functionApp, funcName, funcAppService, pickedFuncs at the same time, and they will be used in the following code

Reply:
I will refactor as much as I can though